task/task-darwinia-ethereum/src/service/ethereum/mod.rs

```diff
@@ -56,104 +57,82 @@ fn get_topics_list(ethereum_config: EthereumConfig) -> Vec<(H160, Vec<H256>)> {
 }
 
 #[derive(Debug)]
-pub struct LikeDarwiniaWithLikeEthereumEthereumScanService {
+pub struct EthereumScanService {
     _greet: Lifeline,
 }
 
-impl BridgeService for LikeDarwiniaWithLikeEthereumEthereumScanService {}
+impl BridgeService for EthereumScanService {}
 
-impl lifeline::Service for LikeDarwiniaWithLikeEthereumEthereumScanService {
+impl lifeline::Service for EthereumScanService {
     type Bus = DarwiniaEthereumBus;
     type Lifeline = anyhow::Result<Self>;
 
     fn spawn(bus: &Self::Bus) -> Self::Lifeline {
         // Receiver & Sender
-        let mut rx = bus.rx::<ToEthereumMessage>()?;
         let sender_to_relay = bus.tx::<ToRelayMessage>()?;
         let sender_to_redeem = bus.tx::<ToRedeemMessage>()?;
-        let sender_to_ethereum = bus.tx::<ToEthereumMessage>()?;
+        let receiver_redeem = bus.rx::<ToRedeemMessage>()?;
 
         // Datastore
         let state = bus.storage().clone_resource::<BridgeState>()?;
 
-        let service_name = format!("{}-service-ethereum-scan", DarwiniaEthereumTask::NAME);
-        let _greet = Self::try_task(&service_name.clone(), async move {
-            let mut is_started = false;
-            while let Some(message) = rx.recv().await {
-                match message {
-                    ToEthereumMessage::Start => {
-                        if is_started {
-                            log::warn!(
-                                target: DarwiniaEthereumTask::NAME,
-                                "The service {} has been started",
-                                service_name.clone()
-                            );
-                            continue;
-                        }
-                        let cloned_state = state.clone();
-                        let cloned_sender_to_relay = sender_to_relay.clone();
-                        let cloned_sender_to_redeem = sender_to_redeem.clone();
-                        let cloned_sender_to_ethereum = sender_to_ethereum.clone();
-                        tokio::spawn(async move {
-                            run(
-                                cloned_state,
-                                cloned_sender_to_relay,
-                                cloned_sender_to_redeem,
-                                cloned_sender_to_ethereum,
-                            )
-                            .await
-                        });
-                        is_started = true;
-                    }
-                    ToEthereumMessage::Restart(force) => {
-                        if force {
-                            is_started = false;
-                        }
-                        let mut cloned_sender_to_ethereum = sender_to_ethereum.clone();
-                        cloned_sender_to_ethereum
-                            .send(ToEthereumMessage::Start)
-                            .await?;
-                    }
-                    _ => continue,
```

Comment:
What's the new restart mechanism if this is deleted?

Reply:
The darwinia/ethereum scan service will auto start with task. all code run in a loop
https://github.com/darwinia-network/bridger/blob/da5fdeeb1b3fd24b8b40eff8a956432ad68ac899/task/task-darwinia-ethereum/src/service/ethereum/mod.rs#L105-L123
when happen an error, will start again

Reply:
Also we can use the kv command to control service state

start darwinia/ethereum scan service
```bash
# set darwinia
./target/release/bridger kv -n task-darwinia-ethereum put scan.darwinia.running true
# set ethereum
./target/release/bridger kv -n task-darwinia-ethereum put scan.ethereum.running true
```

stop darwinia/ethereum scan service
```bash
# set darwinia
./target/release/bridger kv -n task-darwinia-ethereum put scan.darwinia.running false
# set ethereum
./target/release/bridger kv -n task-darwinia-ethereum put scan.ethereum.running false
```